ledger/sandbox-classic/src/main/scala/platform/sandbox/ConfigConverter.scala

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml
+package platform.sandbox
+
+import platform.sandbox.config.SandboxConfig
+
+import ledger.participant.state.kvutils.app.{
+  Config,
+  Mode,
+  ParticipantConfig,
+  ParticipantIndexerConfig,
+  ParticipantRunMode,
+}
+import ledger.sandbox.{BridgeConfig, BridgeConfigProvider}
+import lf.language.LanguageVersion
+import platform.common.LedgerIdMode
+import platform.sandbox.config.SandboxConfig.EngineMode
+import scalaz.syntax.tag._
+import scala.jdk.DurationConverters._
+
+import java.util.UUID
+
+object ConfigConverter {
+  private val DefaultH2SandboxJdbcUrl = "jdbc:h2:mem:sandbox;db_close_delay=-1"
+
+  def toSandboxOnXConfig(
+      sandboxConfig: SandboxConfig,
+      maybeLedgerId: Option[String],
+  ): Config[BridgeConfig] = {
+    val singleCombinedParticipant = ParticipantConfig(
+      mode = ParticipantRunMode.Combined,
+      participantId = sandboxConfig.participantId,
+      shardName = None,
+      address = sandboxConfig.address,
+      port = sandboxConfig.port,
+      portFile = sandboxConfig.portFile,
+      // When missing, sandbox-classic used an in-memory ledger.
+      // For Sandbox-on-X we don't offer that, so default to H2
+      serverJdbcUrl = sandboxConfig.jdbcUrl.getOrElse(DefaultH2SandboxJdbcUrl),
+      managementServiceTimeout = sandboxConfig.managementServiceTimeout,
+      // TODO SoX-to-sandbox-classic: Wire up all indexer configurations
+      indexerConfig = ParticipantIndexerConfig(
+        allowExistingSchema = false,
+        inputMappingParallelism = sandboxConfig.maxParallelSubmissions,
+      ),
+    )
+
+    val extraBridgeConfig = BridgeConfig(
+      conflictCheckingEnabled = true,
+      implicitPartyAllocation = sandboxConfig.implicitPartyAllocation,
+      authService = sandboxConfig.authService,
+      timeProviderType =
+        sandboxConfig.timeProviderType.getOrElse(SandboxConfig.DefaultTimeProviderType),
+      // TODO SoX-to-sandbox-classic: Dedicated submissionBufferSize CLI param for sanbox-classic
+      submissionBufferSize = sandboxConfig.maxParallelSubmissions,
```

Comment:
Not exactly the same thing, but fair enough

Reply:
It's not the same thing, but it's defensive. I wouldn't pollute the `SandboxConfig` with the SoX-specific configurations until they're not specifically needed. wdyt?

Reply:
https://github.com/digital-asset/daml/blob/main/ledger/sandbox-classic/src/main/scala/platform/sandbox/Cli.scala#L53